backend/src/API/coffeeChatAPI.ts

```diff
@@ -145,61 +179,135 @@ export const checkMemberMeetsCategory = async (
     member1.subteams.every((team) => !member2.subteams.includes(team));
+  let status: MemberMeetsCategoryStatus = 'no data';
+  let message: string = '';
 
-  if (category === 'an alumni') {
-    return (await allMembers()).every((member) => member.email !== otherMember?.email);
-  }
-  if (category === 'courseplan member') {
-    return otherMember?.subteams.includes('courseplan');
-  }
-  if (category === 'a pm (not your team)') {
-    return otherMember?.role === 'pm' && submitter && haveNoCommonSubteams(submitter, otherMember);
-  }
-  if (category === 'business member') {
-    return otherMember?.role === 'business';
-  }
-  if (category === 'is/was a TA') {
-    return otherMemberProperties ? otherMemberProperties.ta : undefined;
-  }
-  if (category === 'major/minor that is not cs/infosci') {
-    return otherMemberProperties ? otherMemberProperties.notCsOrInfosci : undefined;
+  // If otherMember doesn't exist in the DB, assume they are an alumni
+  if (!otherMember && category === 'an alumni') {
+    return { status: 'pass', message };
   }
-  if (category === 'idol member') {
-    return otherMember?.subteams.includes('idol');
-  }
-  if (category === 'a newbie') {
-    return otherMemberProperties ? otherMemberProperties.newbie : undefined;
-  }
-  if (category === 'from a different college') {
-    return otherMemberProperties && submitterProperties
-      ? otherMemberProperties.college !== submitterProperties.college
-      : undefined;
-  }
-  if (category === 'curaise member') {
-    return otherMember?.subteams.includes('curaise');
-  }
-  if (category === 'cornellgo member') {
-    return otherMember?.subteams.includes('cornellgo');
-  }
-  if (category === 'a tpm (not your team)') {
-    return otherMember?.role === 'tpm' && submitter && haveNoCommonSubteams(submitter, otherMember);
-  }
-  if (category === 'carriage member') {
-    return otherMember?.subteams.includes('carriage');
-  }
-  if (category === 'qmi member') {
-    return otherMember?.subteams.includes('queuemein');
-  }
-  if (category === 'a lead (not your role)') {
-    return (
-      otherMember?.role === 'lead' &&
-      (submitter?.role !== 'lead'
-        ? otherMemberProperties?.leadType !== submitter?.role
-        : otherMemberProperties?.leadType !== submitterProperties?.leadType)
-    );
-  }
-  if (category === 'cuapts member') {
-    return otherMember?.subteams.includes('cuapts');
+
+  // If otherMember and submitter don't exist, status should stay undefined
+  if (otherMember && submitter) {
+    if (category === 'an alumni') {
+      status = (await allMembers()).every((member) => member.email !== otherMember.email)
+        ? 'pass'
+        : 'fail';
+      status === 'fail' &&
+        (message = `${otherMember.firstName} ${otherMember.lastName} is not an alumni`);
+    } else if (category === 'courseplan member') {
+      status = otherMember.subteams.includes('courseplan') ? 'pass' : 'fail';
+      status === 'fail' &&
```

Comment:
just curious: is this idiomatic? not saying we need to get rid of it. It's possible that I've just not seen this pattern before!

Reply:
Hmm it's probably more idiomatic to use `if`